src/crypto/x509/x509.go

```diff
@@ -1806,6 +1816,16 @@ func buildExtensions(template *Certificate, subjectIsEmpty bool, authorityKeyId
 		n++
 	}
 
+	// This extension is not critical
+	if template.KeyUsage == KeyUsageDigitalSignature && template.IsDC && !oidInExtensions(oidExtensionDelegatedCredential, template.ExtraExtensions) {
+		ret[n].Id = oidExtensionDelegatedCredential
+		ret[n].Value = nil
```

Comment:
I think this is incorrect and we actually want the ASN.1 encoding of nil. Can we confirm that the ASN.1 structure matches the draft and have a test for this? I got this wrong the first time I wrote it, and Digicert got it right, prompting some confusion.

Reply:
You are right. I just added the certificate from the draft for testing, and correctly aded asni.NULL.